LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/text-decoration-thickness-computed.html

```diff
@@ -15,6 +15,23 @@
 test_computed_value("text-decoration-thickness", "auto");
 test_computed_value("text-decoration-thickness", "from-font");
 test_computed_value("text-decoration-thickness", "calc(10px - 8px)", "2px");
+
+test_computed_value("text-decoration-thickness", "32px", "32px");
+test_computed_value("text-decoration-thickness", "2em", "32px");
+test_computed_value("text-decoration-thickness", "200%", "200%");
+
+test_computed_value("text-decoration-thickness", "calc(2em - 0px)", "32px");
+test_computed_value("text-decoration-thickness", "calc(200% - 0px)", "32px");
+test_computed_value("text-decoration-thickness", "calc(0.5em - 0px)", "8px");
+test_computed_value("text-decoration-thickness", "calc(50% - 0px)", "8px");
+
+test_computed_value("text-decoration-thickness", "calc(2em - 8px)", "24px");
+test_computed_value("text-decoration-thickness", "calc(2em - 0.5em)", "24px");
+test_computed_value("text-decoration-thickness", "calc(2em - 50%)", "24px");
+
+test_computed_value("text-decoration-thickness", "calc(200% - 8px)", "24px");
+test_computed_value("text-decoration-thickness", "calc(200% - 0.5em)", "24px");
+test_computed_value("text-decoration-thickness", "calc(200% - 50%)", "150%");
```

Comment:
Actually, I think we should not resolve to a length when mixing percentages and lengths. Firefox & Chrome preserve the `calc()` (although simplifies to the simplest form)

So: `calc(200% - 0px)` becomes 200% , etc.

This makes a difference when you inherit the values. 200% should be relative to the current element always.

Reply:
I referred font-size-computed.html [1] when I wrote this test. But that seems to result in inconsistency between WebKit and Gecko/Blink. I aligned the WebKit's behavior with others.

[1] https://github.com/WebKit/WebKit/blob/78ee0acc6e86f689946238f6f70318a0c3138bbf/LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-size-computed.html#L50